### PR TITLE
Add blog article by Nathan Ellingwood and Siva Rajamanickam

### PR DESCRIPTION
**Pull request by ndellingwood**
# Description

Submission of blog article for consideration by BSSw:

Title: Practices for Productive Software Development using a Stable Performance Portable Ecosystem

Changes to be committed:
	new file:   PracticesPerformancePortableEco.md

EB Member: @bernhold 

Addresses issue: N/A

## PR checklist for files displayed on bssw.io site

* [x] ***[Author]*** `@mention` the BSSw.io editorial board member `@<eb-member-id>` in **Description** above assigned to shepherd your PR.
* [x] ***[Author]*** Add the `<issue-id>` in the **Description** above for the associated GitHub Issue.
* [x] ***[Author]*** Inspect the content in the `*.md` file(s) as rendered in GitHub for this PR.
* [ ] ***[EB Mem]*** Assign this PR to the EB member `<eb-member-id>`.
* [ ] ***[EB Mem]*** Assign this PR to the author of the PR `<pr-author-id>`.
* [ ] ***[EB Mem]*** Add one or more Reviewers.
* [x] ***[EB Mem]*** Add label `content: <content-type>` for the type of contribution.
* [x] ***[EB Mem]*** Add to Project `Content Development` (see [Content Development]).
* [x] ***[EB Mem]*** Add [meta-data] to the `*.md` file(s) (set `Publish: preview`).
* [x] ***[EB Mem]*** Add label `preview` (so PR branch will be merged to 'preview' branch and watch for possible merge failures).
* [ ] ***[EB Mem]*** Rebuild [preview] site and confirm new content is there, renders correctly and is returned in searches.
* [ ] ***[Author]*** Make any final changes to the PR based on feedback.
* [ ] ***[EB Mem]*** Rebuild [preview] site and re-confirm content looks correct.
* [ ] ***[EB Mem]*** Ensure at least one reviewer signs off on the final changes.
* [ ] ***[EB Mem]*** Change meta-data to `Publish: yes` and commit if fully ready to publish.
* [ ] ***[EB Mem]*** Move the PR to "Ready to Publish" in [Content Development].
* [ ] ***[EB Mem]*** Assign to one of the site maintainers to carry out final publication steps.
* [ ] ***[EB Mem]*** Verify that all needed files are present in the PR (article, images, updates to Site/Homepage.md carousel and/or Site/Announcements/Announcements.md as appropriate).
* [ ] ***[EB Mem]*** Merge PR. (Should automatically move to "Done" in [Content Development].)
* [ ] ***[EB Mem]*** Verify new new contribution shows up on [bssw.io] as expected.

NOTE:
* Checklist items prefixed with ***[Author]*** are expected to be performed by the author of the PR or can be performed by the author.
* Checklist items prefixed with ***[EB Mem]*** must be performed by a [BSSw.io Editorial Board (EB) Member](https://betterscientificsoftware.github.io/bssw.io/bssw_members.html).

[preview]: https://preview.bssw.io
[bssw.io]: https://bssw.io
[Content Development]: https://github.com/betterscientificsoftware/bssw.io/projects/3
[BSSw Internal]: https://github.com/betterscientificsoftware/bssw.io/projects/2
[meta-data]: https://betterscientificsoftware.github.io/bssw.io/bssw_styling_common.html#metadata-section
[wikize_refs.py]: https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy

### CODE REVIEW

**Comment by bernhold** [comment]
@ndellingwood we use GitHub profile pics to build our [contributors page](https://bssw.io/items/authors).  I noticed that you have one of GitHubs glyphs as your profile pic.  We do have others like that on our contributors page, but if you wanted to change it, now might be a good time.  Just wanted to let you know.

File: Articles/Blog/PracticesPerformancePortableEco.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Practices for Productive Software Development using a Stable Performance Portable Ecosystem
```
Comment (bernhold) [inline Articles/Blog/PracticesPerformancePortableEco.md:1]: Would you consider a shorter, more direct title, such as "Practices for Productive Software Development in Kokkos"?  I understand the desire to emphasize Kokkos as a performance portability tool, but we're not really talking about the performance portability aspect of Kokkos (except in that it relates to the breadth of testing).  It is really about the development practices used within the Kokkos project.

File: Articles/Blog/PracticesPerformancePortableEco.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+### Testing, testing, testing
+
+As a performance-portable software library we are expected to support a variety of compilers and versions, hardware, and library configurations, for example with the Kokkos Ecosystem regular testing includes (at the time of writing):
+
+- Compilers: GCC (5.3.0 – 10); Clang (4.0-11); Intel (17-19); NVCC (9.2-11); ROCm/HIP (4.2); OneAPI (sycl-nightly tag); IBM/XL (16.1.1)
+- Hardware: x86_64 (various), KNL, ARMv8 ThunderX2, NVIDIA (K80, P100, V100, A100), AMD (Zen CPU, Vega MI25, MI50, MI60, MI100)
+- Backends: Serial, OpenMP, Pthreads, Cuda, HIP (Experimental), OpenMPTarget (Experimental), Sycl (Experimental), HPX
+
+Notice that we test several experimental options and support several older compilers that are years old in order to address the trade-off mentioned above. Total testing coverage requires hours of testing across several computing systems. It would be an unreasonable burden and detriment to coder productivity to require testing the full combinatorial set of supported builds and configurations. To balance developer productivity while preserving software quality and stability, testing of the code base consists of two stages:
```
Comment (bernhold) [inline Articles/Blog/PracticesPerformancePortableEco.md:33]: The end of this sentence says *two* stages, but discussion of three different types of testing follow.  Should it be *three* stages?  Or do you want to distinguish one (release testing?) in some way?

File: Articles/Blog/PracticesPerformancePortableEco.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+#### Nightly testing
+
+  - Nightly testing is automated by Jenkins. Nightly tests significantly expand on pull request testing to fill the missing gaps in coverage, including various configuration options (some costly), older compilers, and experimental builds. More than 200 nightly tests run on all the aforementioned platforms.
+
+#### Release testing
+
+Kokkos and KokkosKernels exist as standalone libraries available on GitHub, and are foundational packages within the Trilinos scientific software library (under the “data services” scope). A new release of the Kokkos Ecosystem requires passing testing within Trilinos, providing additional robustness:
+
+- We do not assume the testing suite is fully comprehensive – testing with Trilinos exposes corner cases that may have been missed in the testing suite.
```
Comment (bernhold) [inline Articles/Blog/PracticesPerformancePortableEco.md:49]: Perhaps "..the *Kokkos* test suite is fully comprehensive..."?

File: Articles/Blog/PracticesPerformancePortableEco.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+- Trilinos testing is assumed complex enough to provide coverage for customer applications.
```
Comment (bernhold) [inline Articles/Blog/PracticesPerformancePortableEco.md:50]: Perhaps "Trilinos testing is assumed to be complex enough to provide coverage for most customer applications."?

**Comment by bernhold** [comment]
@ndellingwood would it be possible for you to review these suggestions sooner rather than later?  I'd like to publish this week.  Thanks

File: Articles/Blog/PracticesPerformancePortableEco.md (hunk 1 above)
Comment (ndellingwood) [reply Articles/Blog/PracticesPerformancePortableEco.md]: Good point and thanks for revision suggestion, addressed by 1a4586007f398b110bd1798dfb1e3ba12325aa18

File: Articles/Blog/PracticesPerformancePortableEco.md (hunk 2 above)
Comment (ndellingwood) [reply Articles/Blog/PracticesPerformancePortableEco.md]: Thanks for revision suggestion, addressed by 1a4586007f398b110bd1798dfb1e3ba12325aa18

File: Articles/Blog/PracticesPerformancePortableEco.md (hunk 4 above)
Comment (ndellingwood) [reply Articles/Blog/PracticesPerformancePortableEco.md]: Thanks for revision suggestion, addressed by 1a4586007f398b110bd1798dfb1e3ba12325aa18

**Comment by ndellingwood** [comment]
@bernhold thank you for the feedback! I pushed updates regarding the suggested updates in SHA 1a4586007f398b110bd1798dfb1e3ba12325aa18

**Comment by karbarz** [comment]
@bernhold 
Edits have been made.
